src/access/AccessRestriction.cpp

```diff
@@ -0,0 +1,270 @@
+    }
+
+    return nullptr;
+}
+
+CHIP_ERROR AccessRestriction::CreateCommissioningEntry(SharedPtr<Entry> entry)
+{
+    if (!entry->IsValid())
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    sCommissioningEntries.push_back(entry);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR AccessRestriction::CreateEntry(size_t * index, const Entry & entry, FabricIndex fabricIndex)
+{
+    if (!entry.IsValid())
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    auto localEntry = MakeShared<Entry>(entry);
+    size_t newIndex;
+
+    localEntry->fabricIndex = fabricIndex;
+
+    if (mFabricEntries.find(fabricIndex) == mFabricEntries.end())
+    {
+        mFabricEntries[fabricIndex] = std::vector<SharedPtr<Entry>>();
+        mFabricEntries[fabricIndex].push_back(localEntry);
+    }
+    else
+    {
+        mFabricEntries[fabricIndex].push_back(localEntry);
+    }
+
+    newIndex = mFabricEntries[fabricIndex].size() - 1;
+
+    for (EntryListener * listener = mListeners; listener != nullptr; listener = listener->mNext)
+    {
+        listener->OnEntryChanged(fabricIndex, newIndex, localEntry, EntryListener::ChangeType::kAdded);
+    }
+
+    if (index != nullptr)
+    {
+        *index = newIndex;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR AccessRestriction::DeleteEntry(size_t index, const FabricIndex fabricIndex)
+{
+    auto entry = GetEntry(fabricIndex, index);
+    if (entry == nullptr)
+    {
+        return CHIP_ERROR_NOT_FOUND;
+    }
+    else
+    {
+        mFabricEntries[fabricIndex].erase(mFabricEntries[fabricIndex].begin() + static_cast<int>(index));
+
+        for (EntryListener * listener = mListeners; listener != nullptr; listener = listener->mNext)
+        {
+            listener->OnEntryChanged(fabricIndex, index, entry, EntryListener::ChangeType::kRemoved);
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR AccessRestriction::UpdateEntry(size_t index, const Entry & entry, const FabricIndex fabricIndex)
+{
+    auto localEntry = GetEntry(fabricIndex, index);
+    if (localEntry != nullptr)
+    {
+        *localEntry = entry;
+
+        for (EntryListener * listener = mListeners; listener != nullptr; listener = listener->mNext)
+        {
+            listener->OnEntryChanged(fabricIndex, index, localEntry, EntryListener::ChangeType::kUpdated);
+        }
+
+        return CHIP_NO_ERROR;
+    }
+
+    return CHIP_ERROR_NOT_FOUND;
+}
+
+CHIP_ERROR AccessRestriction::Check(const SubjectDescriptor & subjectDescriptor, const RequestPath & requestPath)
+{
+    ChipLogProgress(DataManagement, "AccessRestriction: action %d", static_cast<int>(requestPath.requestType));
+
+    if (requestPath.requestType == RequestType::kRequestTypeUnknown)
+    {
+        ChipLogError(DataManagement, "AccessRestriction: RequestPath type is unknown");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // wildcard event subscriptions are allowed. All other requests require an entity id
```

Comment:
```suggestion
    // wildcard event subscriptions are allowed since wildcard is only used when setting up the subscription and we want that request to succeed (when generating the report, this method will be called with the specific event id). All other requests require an entity id
```

Or some comment explaining why the wildcard check here should succeed

Reply:
done